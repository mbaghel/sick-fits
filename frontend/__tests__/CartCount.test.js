import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import CartCountComponent from "../components/CartCount";

describe("<CartCount />", () => {
  it("renders without errors", () => {
    shallow(<CartCountComponent count={10} />);
  });

  it("matches the snapshot", () => {
    const wrapper = shallow(<CartCountComponent count={6} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("handles updates to its props", () => {
    const wrapper = shallow(<CartCountComponent count={25} />);
    expect(toJSON(wrapper)).toMatchSnapshot();

    wrapper.setProps({ count: 19 });
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
