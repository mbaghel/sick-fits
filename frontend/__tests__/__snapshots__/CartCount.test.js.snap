// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount /> handles updates to its props 1`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="25"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        25
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;

exports[`<CartCount /> handles updates to its props 2`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="19"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        19
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;

exports[`<CartCount /> matches the snapshot 1`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="6"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        6
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;
